examples/repeated-compute/src/main.rs

```diff
@@ -0,0 +1,249 @@
+                &OVERFLOW => "(overflow)".to_string(),
+                n => n.to_string(),
+            })
+            .collect::<Vec<String>>();
+        log::info!("Results: {printed_numbers:?}");
+    }
+}
+
+fn generate_rand() -> u16 {
+    let mut bytes = [0u8; 2];
+    getrandom::getrandom(&mut bytes[..]).unwrap();
+    u16::from_le_bytes(bytes)
+}
+
+async fn compute(local_buffer: &mut [u32], context: &WgpuContext) {
+    log::info!("Beginning GPU compute on data {local_buffer:?}.");
+    // Local buffer contents -> GPU storage buffer
+    // Adds a write buffer command to the queue. This command is more complicated
+    // than it appears.
+    context.queue.write_buffer(
+        &context.storage_buffer,
+        0,
+        bytemuck::cast_slice(local_buffer),
+    );
+    log::info!("Wrote to buffer.");
+
+    let mut command_encoder = context
+        .device
+        .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+
+    {
+        let mut compute_pass =
+            command_encoder.begin_compute_pass(&wgpu::ComputePassDescriptor { label: None });
+        compute_pass.set_pipeline(&context.pipeline);
+        compute_pass.set_bind_group(0, &context.bind_group, &[]);
+        compute_pass.dispatch_workgroups(local_buffer.len() as u32, 1, 1);
+    }
+    // We finish the compute pass by dropping it.
+
+    // Entire storage buffer -> staging buffer.
+    command_encoder.copy_buffer_to_buffer(
+        &context.storage_buffer,
+        0,
+        &context.output_staging_buffer,
+        0,
+        context.storage_buffer.size(),
+    );
+
+    // Finalize the command encoder, add the contained commands to the queue and flush.
+    context.queue.submit(Some(command_encoder.finish()));
+    log::info!("Submitted commands.");
+
+    // Finally time to get our results.
+    // First we get a buffer slice which represents a chunk of the buffer (which we
+    // can't access yet).
+    // We want the whole thing so use unbounded range.
+    let buffer_slice = context.output_staging_buffer.slice(..);
+    // Now things get complicated. WebGPU, for safety reasons, only allows either the GPU
+    // or CPU to access a buffer's contents at a time. We need to "map" the buffer which means
+    // flipping ownership of the buffer over to the CPU and making access legal. We do this
+    // with `BufferSlice::map_async`.
+    //
+    // The problem is that map_async is not an async function so we can't await it. What
+    // we need to do instead is pass in a closure that will be executed when the slice is
+    // either mapped or the mapping has failed.
+    //
+    // The problem with this is that we don't have a reliable way to wait in the main
+    // code for the buffer to be mapped and even worse, calling get_mapped_range or
+    // get_mapped_range_mut prematurely will cause a panic, not return an error.
+    //
+    // Using async channels solves this as awaiting the receiving of a message from
```

Comment:
Connor confirmed that calling poll once after submit and map_async is guaranteed to work :+1: 
Let's still convert the channels to non-async standard ones.

Reply:
What do you recommend for the channels? The standard Rust channels use mutexes which are not allowed on WASM. I used futures-intrusive because I knew that worked (because hello-compute does it).

Reply:
I would use flume with the spin feature enabled, but this needs to be an async receive anyway.

@nical on WebGPU, poll is a no-op, so the wait will actually happen in the async_recv.

Reply:
Ok cool flume but what is the reason why flume should be used over futures-intrusive which is declared as a dependency of wgpu anyways?

Reply:
> What do you recommend for the channels? The standard Rust channels use mutexes which are not allowed on WASM. I used futures-intrusive because I knew that worked (because hello-compute does it).

I did not have the wasm restrictions in mind, bummer. I think that this context is valuable to learn so I suggest you keep these channels or use another one if you want but explain in a comment that:
 - For native, we could use standard rust channel in principle as the work happens during poll and the "async-ness" of the channel does not matter.
 - But since standard channels use muteces they wouldn't work with the wasm target. In addition, for the wasm target with the WebGPU backend, poll() is a no-op and the wait actually happens in the `async_receive`.
 - As a result the most portable solution involved using async channels.

Sorry for the roundabout, it turned out to be a learning occasion for me.

Reply:
No you're fine. I'll definitely include a notice though so that people don't understandably get confused.